worker

```diff
@@ -11,62 +52,96 @@ BH_UUID=${BH_UUID}
 EOF
 )
 
-GITEA_REPO="$GITEA_REPOSITORY_ROOT"/"${BH_USER}"/"${BH_REPO}".git
-#echo $GITEA_REPO #DEBUG
+# Canonicalize the GITEA_REPOSITORY_ROOT before changing directories
+# in case it was a relative path.
+GITEA_REPO=$(realpath --canonicalize-existing \
+  "${GITEA_REPOSITORY_ROOT}/${BH_USER}/${BH_REPO}.git")
+echo 1>&2 "GITEA_REPO=${GITEA_REPO}"
 
-sleep 5
+# Do all the work in a private temporary directory.
+WORKDIR=$(mktemp --directory --tmpdir bids-hook.XXXXXXXXXX)
+echo 1>&2 "WORKDIR=${WORKDIR}"
+# We have an ERR trap to account for this:
+# shellcheck disable=SC2164
+cd 1>&2 "$WORKDIR"
+# We have an EXIT trap which makes this code reachable:
+# shellcheck disable=SC2317
+function cleanup() {
+  echo "# cleaning ${WORKDIR}"
+  chmod --recursive u+w "$WORKDIR"
+  rm --recursive "$WORKDIR"
+} 1>&2
+trap cleanup EXIT
 
-cat <<"EOF"
-<!DOCTYPE html>
-<html lang=en>
-<title>a title</title>
-<p>a paragraph</p>
-<pre>
-EOF
-
-(
-WORKDIR="$(mktemp -d --tmpdir bids-hook."$BH_UUID".XXXXXXXXXXX)"
```

Comment:
My thinking is that the BH_UUID is the only "secret" which is needed to view someone else's bids-validator output, because Gitea doesn't have any way to authenticate the static assets. And this output can contain a bunch of metadata about subjects, etc.

I guess we have bigger problems if someone is logged into the server anyway, but just doing `ls /tmp` at the right time, while logged in as _any_ user, would reveal the BH_UUID for the running jobs, and then you can visit the corresponding webpage with the bids-validator output.